nodestream/project/pipeline_definition.py

```diff
@@ -75,29 +108,84 @@ def from_file_data(cls, data, parent_annotations):
+            targets = set(targets) | parent_targets
         return cls(
             name,
             file_path,
-            set(targets),
-            exclude_targets,
-            {**parent_annotations, **annotations},
+            PipelineConfiguration.from_file_data(
+                {
+                    "targets": targets,
+                    "annotations": {**parent_annotations, **annotations},
+                    "exclude_inherited_targets": exclude_targets,
+                },
+            ),
+        )
+
+    @classmethod
+    def from_plugin_data(
+        cls, data, parent_targets: set, parent_annotations
+    ) -> "PipelineDefinition":
+        print(data)
+        name = data.pop("name")
+        exclude_targets = data.pop("exclude_inherited_targets", False)
+        annotations = data.pop("annotations", {})
+        targets = data.pop("targets", [])
+        if not exclude_targets:
+            targets = set(targets) | parent_targets
+        return cls(
+            name,
+            None,
+            PipelineConfiguration.from_file_data(
+                {
+                    "targets": targets,
+                    "annotations": {**parent_annotations, **annotations},
+                    "exclude_inherited_targets": exclude_targets,
+                },
+            ),
         )
 
     def to_file_data(self, verbose: bool = False):
         using_default_name = self.name == self.file_path.stem
-        if using_default_name and not self.annotations and not verbose:
+        if (
+            using_default_name
+            and not self.get_annotations_from_config()
+            and not verbose
+        ):
             return str(self.file_path)
 
         result = {"path": str(self.file_path)}
         if not using_default_name or verbose:
             result["name"] = self.name
-        if self.annotations or verbose:
-            result["annotations"] = self.annotations
-        if self.targets or verbose:
-            result["targets"] = self.targets
+        annotations = self.get_annotations_from_config()
+        targets = self.get_targets_from_config()
+        if targets or verbose:
+            result["targets"] = list(targets)
+        if annotations or verbose:
+            result["annotations"] = annotations
 
         return result
 
+    def to_plugin_file_data(self):
+        result = {"path": str(self.file_path)}
+        result["name"] = self.name
+        annotations = self.get_annotations_from_config()
+        targets = self.get_targets_from_config()
+        if targets:
+            result["targets"] = list(targets)
+        if annotations:
+            result["annotations"] = annotations
+        return result
+
+    def get_annotations_from_config(self):
+        return self.configuration.annotations
+
+    def get_targets_from_config(self):
+        return self.configuration.targets
+
+    def excluded_inherited_targets(self):
+        return self.configuration.exclude_inherited_targets
```

Comment:
Should we make these `@property` so we can exclude the function calls where this is used?